cynic-codegen/src/enum_derive/input.rs

```diff
@@ -10,7 +10,9 @@ pub struct EnumDeriveInput {
 
     pub schema_path: SpannedValue<String>,
     pub query_module: SpannedValue<String>,
-    pub graphql_type: SpannedValue<String>,
+
+    #[darling(default)]
+    pub graphql_type: Option<SpannedValue<String>>,
 
     #[darling(default)]
     pub(super) rename_all: Option<RenameAll>,
```

Comment:
It might be worth adding a function to get the `graphql_type` to this struct, so you don't have to do that `extract_spanned_value` call everywhere.

Perhaps something like this:

```rust
impl EnumDeriveInput {
    pub fn graphql_type_name(&self) -> String {
        self.graphql_type.as_ref().map(|val| *val.clone()).unwrap_or(self.ident.to_string())
    }

    pub graphql_type_span(&self) -> Span {
        use syn::span::Spanned;
        self.graphql_type.as_ref().map(|val| val.span()).unwrap_or(self.ident.span());
    }
}
```

Reply:
Good advice, but wrong report:
```

24 |         self.graphql_type.as_ref().map(|val| *val.clone()).unwrap_or(self.ident.to_string())
   |                                              ^^^^^^^^^^^^ move occurs because value has type `std::string::String`, which does not implement the `Copy` trait

error[E0507]: cannot move out of dereference of `SpannedValue<std::string::String>`
  --> cynic-codegen/src/inline_fragments_derive/input.rs:21:46
   |
21 |         self.graphql_type.as_ref().map(|val| *val.clone()).unwrap_or(self.ident.to_string())
   |                                              ^^^^^^^^^^^^ move occurs because value has type `std::string::String`, which does not implement the `Copy` trait

error[E0507]: cannot move out of dereference of `SpannedValue<std::string::String>`
  --> cynic-codegen/src/input_object_derive/input.rs:41:46
```